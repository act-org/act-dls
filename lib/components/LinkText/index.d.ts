/**
 * @prettier
 */
import * as React from 'react';
import { Props as LinkBaseProps } from '~/components/LinkBase';
export declare type Props = LinkBaseProps;
declare const LinkText: React.FC<Props>;
export default LinkText;
//# sourceMappingURL=index.d.ts.map