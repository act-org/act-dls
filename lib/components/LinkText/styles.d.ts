/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"linkBase", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map