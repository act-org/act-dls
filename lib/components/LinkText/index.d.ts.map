{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/LinkText/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAE/B,OAAiB,EAAE,KAAK,IAAI,aAAa,EAAE,MAAM,uBAAuB,CAAC;AAIzE,oBAAY,KAAK,GAAG,aAAa,CAAC;AAElC,QAAA,MAAM,QAAQ,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAI7B,CAAC;AAEF,eAAe,QAAQ,CAAC"}