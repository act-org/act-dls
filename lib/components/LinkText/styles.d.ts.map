{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/LinkText/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAKH,wBAQG"}