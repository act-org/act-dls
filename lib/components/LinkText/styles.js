"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = _interopRequireDefault(require("../../constants/colors"));

var _makeStyles = _interopRequireDefault(require("../../styles/makeStyles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * @prettier
 */
var _default = (0, _makeStyles["default"])({
  linkBase: {
    '&:hover': {
      textDecoration: 'underline'
    },
    color: _colors["default"].LINK,
    textDecoration: 'none'
  }
});

exports["default"] = _default;