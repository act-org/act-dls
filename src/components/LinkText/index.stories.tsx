/**
 * @prettier
 */

import * as React from 'react';
import { select, text as knobText } from '@storybook/addon-knobs';

import LinkText from '.';

export const text = (): React.ReactElement<any> => (
  <LinkText
    target={select(
      'Target',
      {
        Blank: '_blank',
        Parent: '_parent',
        Self: '_self',
        Top: '_top',
      },
      '_blank',
    )}
    to={knobText('To', 'https://www.act.org')}
  >
    {knobText('Text', 'Click Me')}
  </LinkText>
);

export default {
  component: LinkText,
  parameters: {
    info: {
      text: 'This is the text variant of the Link component.',
    },
  },
  title: 'Atoms/Link',
};
