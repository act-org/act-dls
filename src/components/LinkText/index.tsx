/**
 * @prettier
 */

import * as React from 'react';

import LinkBase, { Props as LinkBaseProps } from '~/components/LinkBase';

import useStyles from './styles';

export type Props = LinkBaseProps;

const LinkText: React.FC<Props> = (props: Props): React.ReactElement<any> => {
  const classes = useStyles();

  return <LinkBase className={classes.linkBase} {...props} />;
};

export default LinkText;
