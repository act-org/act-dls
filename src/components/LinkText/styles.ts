/**
 * @prettier
 */

import COLORS from '~/constants/colors';
import makeStyles from '~/styles/makeStyles';

export default makeStyles({
  linkBase: {
    '&:hover': {
      textDecoration: 'underline',
    },
    color: COLORS.LINK,
    textDecoration: 'none',
  },
});
