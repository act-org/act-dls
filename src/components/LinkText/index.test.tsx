/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import LinkText from '.';

describe('LinkText', () => {
  it('matches the snapshot', () => {
    const tree = render(
      <LinkText target="_blank" to="https://www.act.org">
        <div>children</div>
      </LinkText>,
    );

    expect(tree).toMatchSnapshot();
  });
});
