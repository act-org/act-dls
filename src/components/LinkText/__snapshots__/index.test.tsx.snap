// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkText matches the snapshot 1`] = `
<div>
  <a
    class="makeStyles-linkBase-1"
    href="https://www.act.org"
    target="_blank"
  >
    <div>
      children
    </div>
  </a>
</div>
`;
